tests/test_tpp_backend.py

```diff
@@ -3380,3 +3382,247 @@ def test_ethnicity_from_sus():
         ethnicity_by_group_6=["2", "2"],
         ethnicity_by_group_16=["4", "7"],
     )
+
+
+def _make_patient_with_decision_support_values():
+    """Makes a patient with decision support values for the Electronic Frailty Index."""
+    session = make_session()
+    session.add_all(
+        [
+            Patient(
+                DecisionSupportValue=[
+                    DecisionSupportValue(
+                        AlgorithmType=1,  # Set by TPP
+                        CalculationDateTime="2020-01-01 00:00:00.000",
+                        NumericValue=0.1,
+                    ),
+                    DecisionSupportValue(
+                        AlgorithmType=1,
+                        CalculationDateTime="2020-07-01 00:00:00.000",
+                        NumericValue=0.2,
+                    ),
+                    DecisionSupportValue(
+                        AlgorithmType=1,
+                        CalculationDateTime="2021-01-01 00:00:00.000",
+                        NumericValue=0.3,
+                    ),
+                ]
+            ),
+        ]
+    )
+    session.commit()
+
+
+def test_decision_support_values_with_unsupported_algorithm_value():
+    _make_patient_with_decision_support_values()
+    with pytest.raises(ValueError) as exinfo:
+        StudyDefinition(
+            population=patients.all(),
+            efi=patients.with_these_decision_support_values(
+                "eFI",
+            ),
+        )
+
+    assert str(exinfo.value) == "Unsupported `algorithm` value: eFI"
+
+
+@pytest.mark.parametrize(
+    "returning", ["binary_flag", "date", "number_of_matches_in_period", "numeric_value"]
+)
+def test_decision_support_values_with_max_date(returning):
+    _make_patient_with_decision_support_values()
+    study = StudyDefinition(
+        population=patients.all(),
+        efi=patients.with_these_decision_support_values(
+            "efi",
+            on_or_before="2021-01-01",
+            returning=returning,
+        ),
+    )
+    results = study.to_dicts()
+    if returning == "binary_flag":
```

Comment:
This a just a style thing, but I'm not sure this "case statement" structure makes for particularly readable parameterised tests. One option is to parameterise the expected value too e.g.:
```py
@pytest.mark.parametrize(
    "returning,expected",
    [
        ("binary_flag", "1"),
        ("date", "2021"),
        ("number_of_matches_in_period", "3"),
        ("numeric_value", "0.3"),
    ],
)
def test_decision_support_values_with_max_date(returning, expected):
    ...
```

There's also an `assert_results` helper which would allow you to write the test itself as:
```py
    study = StudyDefinition(
        population=patients.all(),
        efi=patients.with_these_decision_support_values(
            "efi",
            on_or_before="2021-01-01",
            returning=returning,
        ),
    )
    assert_results(study.to_dicts(), efi=[expected])